frontend/src/pages/editRole/EditRole.tsx

```diff
@@ -1,75 +1,132 @@
-         return { ...u, role: role };
-        }
-        return u;
-      });
-      setUsers(updatedUsers);
+    function updateRole(user: UsersListItem, role: UserRole) {
+      //here user is of type User (not UsersListItem), but it seems to work because the needed properties are named the same
+      console.log(user)
+      apiCall.patch(ApiRoutes.USER, {role: role}, {id: user.id}).then((res) => {
+        console.log(res.data);
+        //replace this user in the userlist with the updated one from res.data
+        const updatedUsers = users?.map((u) => {
+          if (u.id === user.id) {
+            return { ...u, role: res.data.role };
+          }
+          return u;
+        });
+        setUsers(updatedUsers?updatedUsers:null);
+      })
     }
 
-    useEffect(() => {
-      //TODO: moet met GET call
-        /*apiCall.get(ApiRoutes.USERS).then((res) => {
-          console.log(res.data)
-          setUsers(res.data)
-        })*/
-      setUsers([
-        {
-          userId: 1,
-          name: "Alice Kornelis",
-          role: "student",
-          email: "test@test.test",
-          url: "test"
-        },
-        {
-          userId: 2,
-          name: "Bob Kornelis",
-          role: "teacher",
-          email: "test@test.test",
-          url: "test"
-        },
-        {
-          userId: 3,
-          name: "Charlie Kornelis",
-          role: "admin",
-          email: "test@test.test",
-          url: "test"
-        }
-      ]);
-      }, []);
+    const onSearch = (values: any) => {
+      setSearched(true);
+      setUsers(null);
+      //search operation here
+      const params = Object.entries(values)
+        .filter(([key, value]) => value !== undefined)
+        .reduce((obj, [key, value]) => ({ ...obj, [key]: value }), {});
+      const queryString = Object.entries(params)
+        .map(([key, value]) => `${key}=${value}`)
+        .join('&');
+      console.log(queryString);
+      apiCall.get(ApiRoutes.USERS,{params:queryString}).then((res) => {
+        console.log(res.data)
+        setUsers(res.data);
+      })
+      
+    };
 
-    if (users === null) {
-        return (
-          <div style={{ width: "100%", height: "100%", display: "flex", justifyContent: "center", alignItems: "center" }}>
-            <Spin
-              tip="Loading..."
-              size="large"
-            />
-          </div>
-        )
-    }
+return (
+  <div style={{padding: "3rem"}}>
+      <Form
+          form={form}
+          name="search"
+          onFinish={onSearch}
+          initialValues={{ remember: true }}
+      >
+          <Row gutter={24} justify="space-between">
+              <Col span={7}>
+                  <Form.Item
+                      name="email"
+                      rules={[
+                          { type: 'email', message: t("editRole.emailError") },
+                      ]}
+                  >
+                      <Input placeholder={t("editRole.email")} />

```

Comment:
Je gebruikt 3 inputs, dit is oke maar werkt niet zo goed voor kleine schermen. Misschien kan je eens naar [Search](https://ant.design/components/input) kijken waar een dropdown in het begin staat en misschien de search knop aan de andere kant. 
<img width="320" alt="Scherm­afbeelding 2024-05-03 om 11 20 22" src="https://github.com/SELab-2/UGent-6/assets/159531937/df222a6e-3b0f-4aed-8a1b-2cd2e562e498">